src/Tools/ExternalAccess/AspNetCore/EmbeddedLanguages/IAspNetCoreRouteEmbeddedLanguageClassifier.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages
+{
+    internal interface IAspNetCoreRouteEmbeddedLanguageClassifier
```

Comment:
@JamesNK This is the type you would implement.  You would do it as:

```c#
[Export(typeof(AspNetCoreRouteEmbeddedLanguageClassifier))]
class ActualAspNetCoreRouteEmbeddedLanguageClassifier : IAspNetCoreRouteEmbeddedLanguageClassifier
{
    // ...
}
```

@jasonmalinowski would james need a particular 'Shared' attribute here?

Reply:
All the ExternalAccess types that are exposed and allowed to be accessed by the partner need to be inside a special `Api` folder and also within a namespace name suffixed with `.Api`. For example, see everything inside https://github.com/dotnet/roslyn/tree/main/src/Workspaces/Core/Portable/ExternalAccess. This will ensure that the EA partner only accesses the types within `X.Y.Z.Api` namespace.

Reply:
@sharwell Do you know if we have a doc somewhere in the repo to guide how to add RestrictedIVT and patterns to follow to ensure the `RestrictedInternalsVisibleToAnalyzer` validates correct usage of restricted internal API surface?